src/stacked_borrows/mod.rs

```diff
@@ -855,11 +726,22 @@ trait EvalContextPrivExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         );
 
         if protect {
+            // We store tags twice, once in global.protected_tags and once in each call frame.
+            // We do this because consulting a single global set in this function is faster
+            // than attempting to search all call frames in the program for the `FrameExtra`
+            // (if any) which is protecting the popped tag.
+            //
+            // This duplication trades off making `end_call` slower to make this function faster. This
+            // trade-off is profitable in practice for a combination of two reasons.
+            // 1. A single protected tag can (and does in some programs) protect thousands of `Item`s.
+            //    Therefore, adding overhead to in function call/return is profitable even if it only
+            //    saves a little work in this function.
```

Comment:
Now that you moved the comment, "this function" references the wrong function.

I am not sure the comment is truly better here. This function is already huge. Might be better to leave the comment where it was, and just point to it from here?